### ISSUE TITLE
[Feature Request] Flatpak version of gamescope?

### ISSUE THREAD

**Issue by xinyazhang** [Plagman/gamescope#183]
gamescope is really useful to provide FXAA support for games that do not have built-in AA options.

However I found it's not so easy to install gamescope into the flatpak version of steam.
Is it possible for you to support a flatpaked gamescope? Thank you!

P.S. I already build gamescope inside flatpak, and tested against a few games. Here is the repo: https://github.com/xinyazhang/flathub/tree/com.valvesoftware.Steam.Utility.gamescope
(However I probably won't have too much time to maintain it so some official support would be great).

**Comment by emersion** [comment]
Your PR has been merged in Flatpak it seems: https://github.com/flathub/com.valvesoftware.Steam.Utility.gamescope

I don't have plans to maintain an official Flatpak for now, but thanks a lot for taking care of it.

**Comment by iMonZ** [comment]
> > I would say that Snapcraft would make much more sense since Snapcrafts are in the PATH. Or how are you executing gamescope?
> 
> flatpaks can also be in `$PATH`, try `file $(command -v <your-favorite-snap>)`, you'll see it's just a shell script that invokes a symlink to the `snap` command. This still then launches a container, and behaves completely differently from the binary you would usually execute - it's all just a band-aid to make snaps look like normal binaries. flatpak doesn't try to hind behind such scripts by default, but you can write them yourself if you like. There's no functional difference between the two from gamescope's perspective (in that I don't think gamescope works behind the barrier, but I came here looking for information on that - either way, I don't want to launch my whole steam process with gamescope).
> 
> The snap vs flatpak discussion is besides the point either way, both exist and both are valid use cases.
> 
> The above extension works pretty well for Steam, curious if something similar can be done for https://flathub.org/apps/details/com.usebottles.bottles (not a request; I'll likely implement this myself if it doesn't exist yet and I find a minute ;) ).

Okay maybe it’s good to put flatpaks into PATH.
You don’t launch the whole steam in GameScope.
You launch steam as a flatpak and from this you launch a game which will be launched from gamescope.
This should make the most sense

**Comment by TLATER** [comment]
>  You launch steam as a flatpak and from this you launch a game which will be launched from gamescope. This should make the most sense

It would, wouldn't it? That's exactly what the extension @xinyazhang wrote allows you to do; See, containers are like a fully separate installation, with their own packages and such. This means that gamescope, even if installed on your host, will not be available to Steam to launch your games with if it's installed as a flatpak/snap.

Both with flatpaks and snaps, you can therefore not launch Steam and then launch gamescope without installing gamescope into the flatpak/snap first, which is impractical without first creating a flatpak/snap for gamescope, and the point of this issue.

In other words, this issue is about doing exactly what you think should be done :) But anyway, let's not derail this issue with discussing what containers even are, if you want to understand this better I'd suggest asking on IRC channels, or just learning about containers directly somewhere else (no need to ping everybody's inboxes with things they already know).

**Comment by Bitwolfies** [comment]
Related issue, is there anyway to use gamescope with flatpak apps? I have it installed both system wide and as a flatpak, but nothing works. If i do the standard gamescope path-to-flatpak it fails because they all use -- commands. Piping doesnt work either.

**Comment by tinywrkb** [comment]
[Here's one approach of using gamescope with existing Flatpak applications](https://github.com/tinywrkb/org.freedesktop.Platform.VulkanLayer.GameScope).
